include/rmm/detail/aligned.hpp

```diff
@@ -68,7 +82,7 @@ constexpr std::size_t align_up(std::size_t v, std::size_t align_bytes) noexcept
 constexpr std::size_t align_down(std::size_t v, std::size_t align_bytes) noexcept
 {
   assert(is_supported_alignment(align_bytes));
-  return align_up(v, align_bytes) - align_bytes;
+  return is_aligned(v, align_bytes) ? v : align_up(v, align_bytes) - align_bytes;
```

Comment:
Also, thinking about it a bit more, rather than having to introduce `is_aligned` so you can reuse `align_up`, `align_down` is actually simpler to implement directly:

```cxx
return v & ~(align_bytes - 1);
```

(which you may have been thinking of in the first place, but put it in the wrong place.)

Reply:
You are correct. I think I was tired when I created this PR. :)